### DISCUSSION TITLE
Azure Container App Registries-Adding Azure Container Registry from other subscription not working

### DISCUSSION THREAD

**Discussion by ranjeetkpGit** [pulumi/pulumi#10680] [Pulumi in Practice]
Hi,
I am creating an azure container app with Pulumi native v1.66.
My container app in azure 'subscription1' and azure container registry is in 'subscription2'.
I am trying to add Registry when creating container app but it throws error
`ERROR: The resource with name 'testacr' and type 'Microsoft.ContainerRegistry/registries' could not be found in subscription 'subscription1'.`
```typescript
var containerApp= new ContainerApp(appName, new ContainerAppArgs
            {
                Configuration = new ConfigurationArgs
                {
                    Dapr = new DaprArgs
                   {
                     AppPort = 80,
                     AppProtocol = "http",
                     Enabled = true,
                   },
                    Ingress = new IngressArgs
                    {
                        External = false,
                        TargetPort = 80
                    },
                    Secrets = new InputList<SecretArgs>
                    {
                        new SecretArgs { Name = "registry-password", Value = "testpassword"}
                    },
                    Registries = new InputList<RegistryCredentialsArgs>   // From other subscription
                    {
                        new RegistryCredentialsArgs
                        {
                            Server ="[testacr.azurecr.io](http://testacr.azurecr.io/)",
                            Username = "testacr",
                            PasswordSecretRef = "registry-password"
                        }
                    }
                }
}
```

**Comment by nimbinatus** [comment]
Is the container registry also in your code, or has it been created by someone else elsewhere?

If you're accessing two different subscriptions, I think you'll need to define 2 providers. The default provider when you use Azure (or any provider, for that matter) will use a single subscription; it can't switch between subscriptions. So, by defining two providers, you can access two different subscriptions. Sam Cogan did a great write-up on that if it helps: https://samcogan.com/deploying-to-more-than-one-azure-subscription-with-pulumi/

In this case, I think you need to have the container registry using a provider that accesses 'subscription2', and then you're creating the container app using the provider that accesses 'subscription1'. You should then be able to get the output from the registry and use it in the definition of the container app.

Does that help answer your question?

**Reply by ranjeetkpGit** [reply]
> Is the container registry also in your code, or has it been created by someone else elsewhere?

Hi @nimbinatus ,
Thanks for reply.

The container registry in different subscription is an existing resource and created by someone else

In workaround I have tried following CLI command to add registry from other subscription to container app and it works

az containerapp registry set -n MyContainerapp -g MyResourceGroup \
    --server MyExistingContainerappRegistry.azurecr.io --username MyRegistryUsername --password MyRegistryPassword

Do we have any similar workaround in Pulumi?

**Reply by ranjeetkpGit** [reply]
Hi @nimbinatus,

Do we have any similar workaround in Pulumi?
